src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -77,24 +77,41 @@ public void setPerson(Person target, Person editedPerson) {
      *
      * @param target       person to be removed.
      * @param editedPerson person to be added.
-     * @param order        order to be removed.
+     * @param order        order to be added.
      */
-    public void setPersonAndDeleteOrder(Person target, Person editedPerson, Order order) {
+    public void setPersonAndAddOrder(Person target, Person editedPerson, Order order) {
         setPerson(target, editedPerson);
-        internalOrderList.remove(order);
+        internalOrderList.add(order);
     }
 
-
     /**
      * Replaces the person {@code target} in the list with {@code editedPerson}.
      *
      * @param target       person to be removed.
      * @param editedPerson person to be added.
-     * @param order        order to be added.
+     * @param order        order to be removed.
      */
-    public void setPersonAndAddOrder(Person target, Person editedPerson, Order order) {
+    public void setPersonAndDeleteOrder(Person target, Person editedPerson, Order order) {
         setPerson(target, editedPerson);
-        internalOrderList.add(order);
+        internalOrderList.remove(order);
+    }
+
+    /**
+     * Replaces the person {@code target} in the list with {@code editedPerson}.
+     *
+     * @param person        person to be removed.
+     *                      // * @param editedPerson  person to be added.
+     * @param orderToDelete order to be removed.
+     * @param orderToAdd    order to be added.
+     */
+
+    public void setPersonAndEditOrder(Person person, Person editedPerson, Order orderToDelete, Order orderToAdd) {
+        requireAllNonNull(person, orderToDelete, orderToAdd);
+        setPerson(person, editedPerson);
+        int index = internalOrderList.indexOf(orderToDelete);
+        internalOrderList.set(index, orderToAdd);
+        internalOrderList.remove(orderToDelete);
+        internalOrderList.add(orderToAdd);
```

Comment:
Don't need these 2 lines cos `orderToDelete` will get gc'ed automatically and `orderToAdd` replaces the previous index; we don't want duplicates.

Reply:
Yep, just saw the error too, fixed!